shared/src/utils/type-state-selector-map.ts

```diff
@@ -0,0 +1,15 @@
+export const typeSelectorMap = {
+    alarmGroup: 'alarmGroups',
+    client: 'clients',
+    hospital: 'hospitals',
+    mapImage: 'mapImages',
+    material: 'materials',
+    patient: 'patients',
+    personnel: 'personnel',
+    simulatedRegion: 'simulatedRegions',
+    transferPoint: 'transferPoints',
+    vehicle: 'vehicles',
+    viewport: 'viewports',
+} as const;
```

Comment:
You could use `satisfies { [key: string]: keyof ExerciseState}` to validate the plurals.

Reply:
I've added a validation that also checks that the key is a valid element type.